pjmedia/src/pjmedia/stream.c

```diff
@@ -1324,16 +1332,30 @@ static pj_status_t put_frame_imp( pjmedia_port *port,
      */
     if (stream->use_ka)
     {
-	pj_uint32_t dtx_duration;
+        if (stream->start_ka_count) {
```

Comment:
since ka already moved here, do we still need the one in stream_create()?

Reply:
I suggest keep sending keepalive if start_ka_count is 0.

Reply:
Okay. But will need to explain it in the doc.

Reply:
The first KA here seems to be sent only after the configured interval time has elapsed (cmiiw), if that's the case, it won't do hole punching (before any real RTP packet is sent), so I think the one in stream_create() should still be sent even when start_ka_count==0, its main purpose is hole punching not purely KA.

Reply:
oops sorry, it should be `even when start_ka_count != 0`. Hole punching should be done as early as possible, and it does exactly that, IMO.

Btw, the PR title/desc should mention about hole punching too somewhere perhaps, as the main purpose seems about hole punching instead of KA (just piggyback-ing KA mechanism)?